emukit/bayesian_optimization/acquisitions/expected_improvement.py

```diff
@@ -73,3 +73,69 @@ def evaluate_with_gradients(self, x: np.ndarray) -> Tuple:
     def has_gradients(self) -> bool:
         """Returns that this acquisition has gradients"""
         return isinstance(self.model, IDifferentiable)
+
+
+
+class IntegratedExpectedImprovement(Acquisition):
+
+    def __init__(self, model: Union[IModel, IDifferentiable, IPriorHyperparameters], jitter: np.float64 = np.float64(0)) -> None:
+        """
+        This acquisition computes for a given input the improvement over the current best observed value in
+        expectation. This function integrates over hyper-parameters the model by computing the  average of the
+        expected improvements for all samples. For more information see:
+
+        Efficient Global Optimization of Expensive Black-Box Functions
+        Jones, Donald R. and Schonlau, Matthias and Welch, William J.
+        Journal of Global Optimization
+
+        :param model: model that is used to compute the improvement.
+        :param jitter: parameter to encourage extra exploration.
+        """
+
+        self.model = model
+        self.jitter = jitter
+
+    def evaluate(self, x: np.ndarray) -> np.ndarray:
+        """
+        Computes the integrated Expected Improvement with respect to the hyper-parameters of the model. Averages the
+        improvement for all the samples.
+
+        :param x: points where the acquisition is evaluated.
+        """
+
+        if x.ndim == 1: x = x[None, :]
+        num_samples = len(self.model.hyperparameters_samples)
+        improvement = 0
+
+        for sample in self.model.hyperparameters_samples:
+            self.model.fix_model_hyperparameters(sample)
+            acquisition = ExpectedImprovement(self.model, self.jitter)
+            improvement += acquisition.evaluate(x)
+
+        return improvement/num_samples
+
+    def evaluate_with_gradients(self, x: np.ndarray) -> Tuple:
```

Comment:
This method has some repetition compared to `evaluate` above. Would it be possible to reuse some code?

Reply:
that's a good point but it is generic to how all the acquisitions are implemented at the moment. I'll open an issue to correct this in a separated PR.

Reply:
Fair enough, let's tackle this separately

Reply:
In previous examples we made evaluate call the evaluate_with_gradients and only return first argument. It makes sense if one assumes that evaluate with gradients is the thing you usually call.